spec/python/test_eof_exception_u4.py

```diff
@@ -3,8 +3,15 @@
 
 from eof_exception_u4 import EofExceptionU4
 
+# Python 2 has only assertRaisesRegexp method
+# Python 3.11 and below have both assertRaisesRegex and assertRaisesRegexp
+# Python 2.12 and above has only assertRaisesRegex
+# The tests run on both Python 2 & 3, so we add missing method for compatibility
+if not hasattr(unittest.TestCase, 'assertRaisesRegex'):
+    setattr(unittest.TestCase, 'assertRaisesRegex', unittest.TestCase.assertRaisesRegexp)
```

Comment:
Sounds good, appreciate your fix here!

Do you think we can actually extract this into a separate file not to repeat these lines everywhere?

Something akin to [helpers/javascript](https://github.com/kaitai-io/kaitai_struct_tests/tree/master/helpers/javascript)?

Reply:
I was thinking about building a base class for tests, but since I don't test it (I'm too lazy to set up all the language environments that will only be needed for one thing), I was not sure that this will work and I made the simplest solution.

Something like this, but I'm not sure if constructor should have some arguments:
```python
class BaseTest(unittest.TestCase):
  def __init__():
    if not hasattr(unittest.TestCase, 'assertRaisesRegex'):
      setattr(unittest.TestCase, 'assertRaisesRegex', unittest.TestCase.assertRaisesRegexp)
```